model/traceid.go

```diff
@@ -34,22 +35,24 @@ func (t TraceID) ToHex() string {
 	)
 }
 
-// Empty returns if TraceID has zero value
+// Empty returns if TraceID has zero value.
 func (t TraceID) Empty() bool {
 	return t.Low == 0 && t.High == 0
 }
 
-// UnmarshalJSON custom JSON deserializer
+// UnmarshalJSON custom JSON deserializer to retrieve the traceID from the hex
+// encoded representation.
 func (t *TraceID) UnmarshalJSON(traceID []byte) error {
```

Comment:
I don't understand json encoding here? What is this?

Reply:
You'll notice a couple of these UnmarshalJSON and MarshalJSON methods inside the model package. They handle easy conversion between JSON representation and Go struct model. 

This way standard Go serialization is able to take the Go struct payload and marshal into correct Zipkin V2 Span and back. (think traceid struct high64/low64 to hex and (parent)spanid 64bit uint to hex)